src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/domain/price/Price.kt

```diff
@@ -58,7 +58,7 @@ data class Price(
   val effectiveYear: Int,
 ) {
 
-  @OneToMany(mappedBy = "price", fetch = FetchType.EAGER, cascade = [CascadeType.ALL])
+  @OneToMany(mappedBy = "price", fetch = FetchType.EAGER, cascade = [CascadeType.ALL], orphanRemoval = true)
```

Comment:
So this was a subtle issue I came across.  The Spring JPA tests did not highlight this as an issue.  I noticed this when I was running the application locally.

I will be adding an integration test so this will provide coverage for this.